### PR TITLE
feat(waf): add a new parameter charging_mode in cloud domain

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccWafDomainV1_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccWafDomainV1_basic -timeout 360m -parallel 4
=== RUN   TestAccWafDomainV1_basic
=== PAUSE TestAccWafDomainV1_basic
=== CONT  TestAccWafDomainV1_basic
--- PASS: TestAccWafDomainV1_basic (83.74s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       83.856s

make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccWafDomainV1_policy'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccWafDomainV1_policy -timeout 360m -parallel 4
=== RUN   TestAccWafDomainV1_policy
=== PAUSE TestAccWafDomainV1_policy
=== CONT  TestAccWafDomainV1_policy
--- PASS: TestAccWafDomainV1_policy (82.78s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       82.910s
```

### CODE REVIEW

File: huaweicloud/services/waf/resource_huaweicloud_waf_domain.go [hunk 1]
```diff
@@ -102,6 +104,16 @@ func ResourceWafDomainV1() *schema.Resource {
...
 				Default:  false,
 			},
+			"charging_mode": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Default:     PaidType,
+				Description: "schema: Internal",
+				ValidateFunc: validation.StringInSlice([]string{
+					"prePaid",
```
Comment (ShiChangkuo) [inline huaweicloud/services/waf/resource_huaweicloud_waf_domain.go]:
```suggestion
					PaidType,
```

File: huaweicloud/services/waf/resource_huaweicloud_waf_domain.go [hunk 2]
```diff
@@ -14,6 +14,8 @@ import (
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
 )
 
+var PaidType = "prePaid"
```
Comment (ShiChangkuo) [inline huaweicloud/services/waf/resource_huaweicloud_waf_domain.go:17]: it's better to define it as **[]string**

File: huaweicloud/services/waf/resource_huaweicloud_waf_domain.go (hunk 1 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/waf/resource_huaweicloud_waf_domain.go]: fixed

File: huaweicloud/services/waf/resource_huaweicloud_waf_domain.go (hunk 2 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/waf/resource_huaweicloud_waf_domain.go:17]: var PaidType indicates the default value, so I think string is better

**Pull request merged**